homeassistant/components/utility_meter/select.py

```diff
@@ -211,7 +223,7 @@ async def async_next_tariff(self):
             or (state := self.hass.states.get(self.tracked_entity_id)) is None
         ):
             return
-        tariffs = state.attributes.get(ATTR_OPTIONS)
+        tariffs = state.attributes[ATTR_OPTIONS]
```

Comment:
Slight code change here, but if it was optional it would have failed two lines down on `tariffs.index`

Reply:
I think we should just remove this Legacy entity altogether :) it was due to be removed in 2022.7

Reply:
I'm not sure I feel confortable to do the removal.
I'll mark this as draft for now, and maybe you can do the removal before I rebase?

Reply:
PR open at https://github.com/home-assistant/core/pull/76480